test/e2e/framework/service/jig.go

```diff
@@ -378,12 +366,12 @@ func (j *TestJig) SanityCheckService(svc *v1.Service, svcType v1.ServiceType) {
 		if svc.Spec.ExternalName != "" {
 			framework.Failf("unexpected Spec.ExternalName (%s) for service, expected empty", svc.Spec.ExternalName)
 		}
-		if svc.Spec.ClusterIP != api.ClusterIPNone && svc.Spec.ClusterIP == "" {
-			framework.Failf("didn't get ClusterIP for non-ExternamName service")
+		if svc.Spec.ClusterIP == "" {
+			framework.Failf("didn't get ClusterIP for non-ExternalName service")
 		}
 	} else {
 		if svc.Spec.ClusterIP != "" {
-			framework.Failf("unexpected Spec.ClusterIP (%s) for ExternamName service, expected empty", svc.Spec.ClusterIP)
+			framework.Failf("unexpected Spec.ClusterIP (%s) for ExternalName service, expected empty", svc.Spec.ClusterIP)
```

Comment:
Going along with my other comment, if we are going to include within the jig create* methods (which would be good), we should rewrite this to return errors instead of making a decision for the user.
i.e.,
```go
if svc.Spec.ClusterIP != "" {
	return fmt.Errorsf("unexpected Spec.ClusterIP (%s) for ExternamName service, expected empty", svc.Spec.ClusterIP)
}
```